### ISSUE TITLE
Allow an empty value for StringToString

### ISSUE THREAD

**Issue by kmeaw** [spf13/pflag#312]
https://play.golang.org/p/8mP9OQAxndt

The user is not allowed to specify an empty map to --string-to-string option.
We should treat an empty string (or some other special value) as a valid value.
That would give the user an ability to override a default non-empty map.

**Comment by falzm** [comment]
Yes please 🙏

**Comment by NetanelBollag** [comment]
Yes we need it!

**Comment by tomasaschan** [comment]
This issue has not been updated in a long time. Is it still relevant?

(If I receive no answer before the end of the summer, I'll close it. Feel free to re-open if you come here after that happens and it is relevant to you - but if you do, please also comment and specify what outcome you're interested in!)

**Comment by kmeaw** [comment]
Yes, the issue is still relevant - the recent version (v1.0.7) of pflag still treats an empty string as a non-valid value for StringToString.

**Comment by brandon1024** [comment]
Not a bad idea! I can take a look to see what kind of effort would be involved, since I've already been working in that area. I don't imagine it'll take much effort.

**Comment by happysnaker** [comment]
Follow-up: I opened #486 for this. The change keeps omission semantics intact, but lets an explicit `--flag=` clear `StringToString` to a non-nil empty map so callers can override non-empty defaults.

**Comment by tomasaschan** [comment]
See my comment on #486: It's virtually impossible to come up with new behavior here that doesn't break existing behavior in some way. Fixing this is _not in scope for 1.x_.
